### PR TITLE
feat(schematic): fds 828 generate manifest endpoint

### PR DESCRIPTION
**Pull request by andrewelamb**
- Fixes [fds-828](https://sagebionetworks.jira.com/browse/FDS-828)
- Adds the generate google sheet manifest endpoint
- Adds step in schematic specific ci workflow that daves google creds

See initial PR: https://github.com/Sage-Bionetworks/sage-monorepo/pull/2480

### CODE REVIEW

File: libs/schematic/api-description/build/openapi.yaml [hunk 1]
```diff
@@ -753,6 +753,45 @@ paths:
...
+      tags:
+        - ManifestGeneration
+      summary: Generates a list of google sheet links
+      description: Generates a list of google sheet links
+      operationId: generateGoogleSheetManifests
+      parameters:
+        - $ref: '#/components/parameters/schemaUrl'
+        - $ref: '#/components/parameters/addAnnotations'
+        - $ref: '#/components/parameters/datasetIdArray'
+        - $ref: '#/components/parameters/assetViewIdQuery'
```
Comment (linglp) [inline libs/schematic/api-description/build/openapi.yaml:767]: When generating a manifest, assetViewId is not required. For people who are generating a new manifest (without pulling any existing manifests), they don't need an asset view.

Comment (andrewelamb) [reply libs/schematic/api-description/build/openapi.yaml:767]: You want to be an optional parameter correct?

**Review by linglp — CHANGES_REQUESTED** [review]
I could generate an existing manifest using the endpoint.. but could you turn `assetViewId` needs to be optional for generating a new manifest. A new manifest is generated based on the data model, so it doesn't need to interact with synapse.

File: libs/schematic/api-description/build/openapi.yaml (hunk 1 above)
Comment (linglp) [reply libs/schematic/api-description/build/openapi.yaml:767]: Yes. It needs to be optional

**Comment by andrewelamb** [comment]
@linglp I made assetViewId optional

**Comment by andrewelamb** [comment]
@linglp Anything else on this PR?

File: apps/schematic/api/schematic_api/openapi/openapi.yaml [hunk 2]
```diff
@@ -1082,6 +1082,104 @@ paths:
...
+        style: form
+      - description: An array of dataset ids
+        explode: true
+        in: query
+        name: datasetIdArray
+        required: false
+        schema:
+          $ref: '#/components/schemas/DatasetIdArray'
+        style: form
+      - description: "If making one manifest, the title of the manifest If making\
```
Comment (linglp) [inline apps/schematic/api/schematic_api/openapi/openapi.yaml:1115]: Maybe missing a `.` here

File: libs/schematic/api-description/build/openapi.yaml [hunk 3]
```diff
@@ -753,6 +753,50 @@ paths:
...
+        - name: useStrictValidation
+          in: query
+          description: If true, users are blocked from entering incorrect values. If false, users will get a warning when using incorrect values.
+          required: false
+          schema:
+            type: boolean
+            default: true
+        - name: generateAllManifests
+          in: query
+          description: If true, a manifest for all components will be generated, datasetIds will be ignored If false, manifests for each id in datasetIds will be generated
```
Comment (linglp) [inline libs/schematic/api-description/build/openapi.yaml:784]: same as above. I think you missed a `.` after `ignored`

File: apps/schematic/api/schematic_api/openapi/openapi.yaml [hunk 4]
```diff
@@ -1082,6 +1082,104 @@ paths:
       tags:
       - Schema
       x-openapi-router-controller: schematic_api.controllers.schema_controller
+  /generateGoogleSheetManifests:
+    get:
+      description: Generates a list of google sheet links
+      operationId: generate_google_sheet_manifests
+      parameters:
+      - description: The URL of a schema in jsonld form
```
Comment (linglp) [inline apps/schematic/api/schematic_api/openapi/openapi.yaml:1090]: After the schema refactor, the schema url no longer has to be in jsonld form. After you update to use schematic `v24.1.1`,  the schema url could also be a link to the data model csv.

**Review by linglp — APPROVED** [review]
The following tests passed: 
1) generated multiple new manifests (Patient, Biospecimen) using the example data model in jsonld format; 
2) generated an existing manifest using the example data model in jsonld format; 
3) generated an existing manifest when `addAnnotations=true` using the example data model in jsonld format; 
4) generated new manifests when `generateAllManifests=true` using the example data model in jsonld format; 

After update the schematic version to v24.1.1, please be sure to test the endpoint/make changes and ensure that `schema_url` could take a csv link, like this: https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.csv

**Pull request merged**